### PR TITLE
Dockerfile for druid - fixes to get Druid to build

### PR DESCRIPTION
**Pull request by afsheenb**
As issue #3 notes, the Dockerfile in this repo wasn't building; I've been trying to update it to fix any issues that were causing the build to error out. I'm now able to build and am going to move on to testing Druid now in our staging environment. Notes on my changes:

1) I've bundled all of the apt-get install statements together so it only needs to be run once;
2) I've wrapped the java commands that install dependencies and set up maven repositories inside of a /bin/bash -c command, as it makes escaping control characters much easier (my build was dying earlier here, as apparently escaping the \'s in that step wasn't working as exxpected)
3) I've changed the syntax of the mysql database creation command to be cleaner and not rely on |, and fixed another error I was getting re: mysql syntax;
4) Instead of setting a specific version to download as an ENV, I'm just fetching the master branch of metamx/druid from github.

Take a look and let me know if you have any thoughts or questions for me.

Cheers,

--afsheen

### CODE REVIEW

**Comment by xvrl** [comment]
Thanks for the fixes @afsheenb, let me have a look!

File: Dockerfile [hunk 1]
```diff
@@ -6,16 +6,9 @@ RUN apt-get install -y software-properties-common
...
 RUN echo oracle-java-8-installer shared/accepted-oracle-license-v1-1 select true | /usr/bin/debconf-set-selections \
-      && apt-get install -y oracle-java8-installer \
-      && apt-get install -y oracle-java8-set-default
-
-# MySQL (Metadata store)
-RUN apt-get install -y mysql-server
-
-# Supervisor
-RUN apt-get install -y supervisor
+      && apt-get install -y oracle-java8-installer oracle-java8-set-default mysql-server supervisor git
```
Comment (xvrl) [inline Dockerfile]: is there a particular benefit to run everything in one command?
I was mainly going for readability here.

File: Dockerfile [hunk 2]
```diff
@@ -38,52 +28,43 @@ RUN adduser --system --group --no-create-home druid \
...
-RUN git clone -q --branch $DRUID_VERSION --depth 1 https://github.com/metamx/druid.git /tmp/druid
+ADD https://api.github.com/repos/metamx/druid/git/refs/heads/master druid-version.json
+RUN git clone -q --branch master --depth 1 https://github.com/metamx/druid.git /tmp/druid
 WORKDIR /tmp/druid
 # package and install Druid locally
-RUN mvn -U -B versions:set -DnewVersion=$DRUID_VERSION \
-  && mvn -U -B clean install -DskipTests=true -Dmaven.javadoc.skip=true \
-  && cp services/target/druid-services-$DRUID_VERSION-selfcontained.jar /usr/local/druid/lib
+RUN mvn -U -B clean install -DskipTests=true -Dmaven.javadoc.skip=true \
+  && cp services/target/druid-services-0.7.2-SNAPSHOT-selfcontained.jar /usr/local/druid/lib
```
Comment (xvrl) [inline Dockerfile]: hard-coding the version is going to break the next time the pom version changes (as it just did a few hours ago)

File: Dockerfile [hunk 3]
```diff
@@ -38,52 +28,43 @@ RUN adduser --system --group --no-create-home druid \
...
 RUN mvn dependency:get -Dartifact=io.druid:druid-services:0.7.1.1
 
-# Druid (release tarball)
-#ENV DRUID_VERSION 0.7.1.1
-#RUN wget -q -O - http://static.druid.io/artifacts/releases/druid-services-$DRUID_VERSION-bin.tar.gz | tar -xzf - -C /usr/local
-#RUN ln -s /usr/local/druid-services-$DRUID_VERSION /usr/local/druid
 
 # Druid (from source)
 RUN mkdir -p /usr/local/druid/lib /usr/local/druid/repository
-ENV DRUID_VERSION master
```
Comment (xvrl) [inline Dockerfile]: was there an issue with having a variable for this?

File: Dockerfile [hunk 4]
```diff
@@ -38,52 +28,43 @@ RUN adduser --system --group --no-create-home druid \
...
-      -Ddruid.extensions.localRepository=/usr/local/druid/repository \
-      -Ddruid.extensions.remoteRepositories=[\"file:///root/.m2/repository/\",\"https://repo1.maven.org/maven2/\"] \
-      -cp /usr/local/druid/lib/* \
-      io.druid.cli.Main tools pull-deps
 
-WORKDIR /
-
-# Setup metadata store
-RUN /etc/init.d/mysql start && echo "GRANT ALL ON druid.* TO 'druid'@'localhost' IDENTIFIED BY 'diurd'; CREATE database druid CHARACTER SET utf8;" | mysql -u root && /etc/init.d/mysql stop
+RUN /bin/bash -c 'java "-Ddruid.extensions.coordinates=[\"io.druid.extensions:druid-s3-extensions\",\"io.druid.extensions:mysql-metadata-storage\"]" -Ddruid.extensions.localRepository=/usr/local/druid/repository -Ddruid.extensions.remoteRepositories=[\"file:///root/.m2/repository/\",\"https://repo1.maven.org/maven2/\"] -cp /usr/local/druid/lib/* io.druid.cli.Main tools pull-deps'
```
Comment (xvrl) [inline Dockerfile]: I find the bash -c version hard to read. Can you clarify what was not working with the current command?

Comment (xvrl) [inline Dockerfile]: having it on multiple lines also makes it easier to edit

**Comment by fjy** [comment]
@xvrl what happened with this PR? It has been outstanding since May 2015

**Comment by xvrl** [comment]
@fjy I've never heard back after making comments. Lots of things have changed since, so I'm going to close this. @afsheenb please reopen if you'd like to add anything.

**Pull request closed without merging**